### DISCUSSION TITLE
How to deal with handleSubmit @typescript-eslint/no-misused-promises error?

### DISCUSSION THREAD

**Discussion by vnugent** [react-hook-form/react-hook-form#10855] [Q&A]
I recently adopted `ts-standard` (the Typescript version of https://standardjs.com/) and it's flagging a lot of errors at form `onSubmit()`, which is correct because onSubmit is expecting a handler function returning void, but `handleSubmit()` returns Promise<void>.   How would you fix these errors?  Like my fix below?

Error: 
**Promise-returning function provided to attribute where a void return was expected. (@typescript-eslint/no-misused-promises)**

```javascript
<form
   onSubmit={handleSubmit(submitHandler)}
>
```

My fix
```javascript
<form
   onSubmit={() => { void handleSubmit(submitHandler) }}
>
```

**Comment by vnugent** [comment]
Similar discussion https://github.com/orgs/react-hook-form/discussions/10965

**Comment by escapedcat** [comment]
Thanks! This also might be related: https://github.com/orgs/react-hook-form/discussions/8020
